### DISCUSSION TITLE
Demos using real datasets

### DISCUSSION THREAD

**Discussion by kevinnkansah** [koaning/embetter#98] [Ideas]
Could someone please make demos on how to use the Encoders correctly on an actual dataset like [(emotions dataset)](https://www.kaggle.com/datasets/nelgiriyewithana/emotions) for example because I am struggling to use SequenceEncoder. Noob in NLP btw

**Comment by koaning** [comment]
I am not behind a laptop now, but could you elaborate on the sequence encoder? I dont recall a component with that name in this repository.

**Reply by kevinnkansah** [reply]
I meant SentenceEncoder my bad, 

```
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from embetter.text import SentenceEncoder
from embetter.grab import ColumnGrabber
from sklearn.pipeline import make_pipeline
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, confusion_matrix
pd.set_option('display.max_colwidth', None)
df = pd.read_csv("data/text.csv")

df.drop("Unnamed: 0", axis=1, inplace=True)
df = df.sample(250)
X=df.drop(['label'],axis=1)
y=df['label']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
embedding_pipeline = make_pipeline(
    ColumnGrabber("text"),
    SentenceEncoder(name="all-MiniLM-L6-v2", num_threads=2),
)
model = make_pipeline(
    embedding_pipeline,
    GradientBoostingClassifier(n_estimators=100, learning_rate=1.0)
)
model
model.fit(X_train, y_train)
y_pred = model.predict(X_test)
model.score(X_test, y_test)
cm = confusion_matrix(y_pred=y_pred, y_true=y_test)
sns.heatmap(cm)
plt.show()
```
i feel like i am missing a step in my pipeline, could you please help

**Reply by koaning** [reply]
It looks ok when I glance at this code, do you receive an error message?

**Reply by kevinnkansah** [reply]
No there are no syntax or runtime errors but i think some sort of logic error. I get a model score of **0**, but i restarted my notebook and it was fine.

**Reply by koaning** [reply]
Could you share the confusion matrix?

**Reply by kevinnkansah** [reply]
![Screenshot from 2024-04-06 17-03-39](https://github.com/koaning/embetter/assets/144918669/7e692484-22f6-496d-9225-c2aa23f41594)
I believe it is like that but I accidently made the target variable the training variable, so this is my fault
![Screenshot from 2024-04-06 17-27-14](https://github.com/koaning/embetter/assets/144918669/d93d9846-95e6-4416-9c48-03c0e1586b25)
This is it after I restarted my notebook, also I am sampling only 250 data entries instead of 416809 in the original dataset but it takes forever even just to do 500 data entries , my score is also 0.34 btw

Link to dataset: ![Emotions Dataset](https://www.kaggle.com/datasets/nelgiriyewithana/emotions)

**Reply by koaning** [reply]
embeddings can be slow yeah but thats something i cant fix directly because the underlying models carry heavy compute workloads.

**Comment by kevinnkansah** [comment]
Here is one but with 10000 data entries
![Screenshot from 2024-04-06 17-38-28](https://github.com/koaning/embetter/assets/144918669/9e7892ec-5bd5-43be-b23f-e78b5c85dbfd)
